src/routes/Apps/App/Index.tsx

```diff
@@ -33,26 +33,38 @@ import './Index.css'
 // this would eliminate the use of page title strings as navigation keys and instead use the url
 
 interface ComponentState {
+    loading: boolean
     validationErrors: string[];
     packageId: string;
 }
 
 class Index extends React.Component<Props, ComponentState> {
     state: ComponentState = {
+        loading: false,
```

Comment:
Had more or less the same solution, but used a "modelLoaded" with initialization state of "false" rather then "loading" flag.  This avoids the double set state call.  At class is recreated on navigation, one doesn't have to reset every time